acncli/installer/install.go

```diff
@@ -0,0 +1,111 @@
+package installer
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	c "github.com/Azure/azure-container-networking/acncli/api"
+)
+
+type InstallerConfig struct {
+	SrcDir         string
+	DstBinDir      string
+	DstConflistDir string
+	IPAMType       string
+	ExemptBins     map[string]bool
+	OSType         string
+	CNITenancy     string
+	CNIMode        string
+}
+
+func (i *InstallerConfig) SetExempt(exempt []string) {
+	// set exempt binaries to skip installing
+	// convert to all lower case, strip whitespace, and split on comma
+	for _, binName := range exempt {
+		i.ExemptBins[binName] = true
+	}
+}
+
+func (i *InstallerConfig) SetOSType(osType string) error {
+	if strings.EqualFold(osType, c.Linux) || strings.EqualFold(osType, c.Windows) {
+		i.OSType = fmt.Sprintf("%s_%s", osType, c.Amd64)
+	} else {
+		return fmt.Errorf("Invalid target OS supplied: %s", osType)
+	}
+	return nil
+}
+
+func (i *InstallerConfig) SetCNIType(cniType string) error {
+	// get paths for singletenancy and multitenancy
+	switch {
+	case strings.EqualFold(cniType, c.Multitenancy):
+		i.CNITenancy = fmt.Sprintf("%s-%s", c.CNI, c.Multitenancy)
+	case strings.EqualFold(cniType, c.Singletenancy):
+		i.CNITenancy = c.CNI
+	default:
+		return fmt.Errorf("No CNI type supplied, please use %q or %q and try again", c.Transparent, c.Bridge)
+	}
+	return nil
+}
+
+func (i *InstallerConfig) SetCNIDatapathMode(cniMode string) error {
+	// get paths for singletenancy and multitenancy
+	if cniMode != "" {
+		if strings.EqualFold(cniMode, c.Transparent) || strings.EqualFold(cniMode, c.Bridge) {
+			i.CNIMode = cniMode
+			return nil
+		}
+
+		return fmt.Errorf("No CNI mode supplied, please use %q or %q and try again", c.Transparent, c.Bridge)
+	}
+	return nil
+}
+
+func InstallLocal(installerConf InstallerConfig) error {
+	fmt.Printf("📁 - Checking if destination bin directory (%s) exists...\n", installerConf.DstBinDir)
```

Comment:
<!comment> Great idea with the emojis, are these printed in stdout or logfile ? If logfile are we sure the form exists ? If stdout, some terminals may not have support for emojis.